File: EasyTownHall/Backend/src/main/java/de/othr/easytownhall/models/dtos/ApplicationInstanceDTO.java
package de.othr.easytownhall.models.dtos;

import lombok.Data;

import java.time.LocalDate;
import java.util.List;

@Data
public class ApplicationInstanceDTO {
    private Long id;
    private LocalDate submissionDate;
    private String status;
    private String applicationFormName;
    private List<String> filledFormData; // Liste der Formulardaten
}

File: EasyTownHall/Frontend/src/app/models/DTOs/ApplicationInstanceDTO.ts
export interface ApplicationInstanceDTO {
    id: number; // Entspricht Long in Java
    submissionDate: string; // Verwende string für ISO-Datum, alternativ Date
    status: string; // Status als String
    user: {id: number, email: string};
    applicationFormName: string; // Name des Formulars
    filledFieldDTOS: {
        data: string,
        fieldName: string,
        fieldType: string,
        required: boolean
    }[]; // Liste der Formulardaten als Array von Strings
}
  